Sources/CircularProgress/CustomButton.swift

```diff
@@ -77,8 +85,16 @@ extension CALayer {
 		animation.duration = duration
 		animation.fillMode = .forwards
 		animation.isRemovedOnCompletion = false
-		add(animation, forKey: keyPath)
-		setValue(color, forKey: keyPath)
+		add(animation, forKey: keyPath) { [weak self] _ in
+			self?.setValue(color, forKey: keyPath)
+		}
+	}
+
+	func add(_ animation: CAAnimation, forKey key: String?, completion: ((Bool) -> Void)?) {
```

Comment:
Actually, no point in making the closure optional as then it overlaps with the built-in method.

```suggestion
	func add(_ animation: CAAnimation, forKey key: String?, completion: ((Bool) -> Void)) {
```

Reply:
Good catch.